spikeinterface/extractors/mdaextractors.py

```diff
@@ -0,0 +1,221 @@
+from spikeinterface.core import BaseRecording, BaseRecordingSegment, BaseSorting, BaseSortingSegment
+from spikeinterface.core.core_tools import write_binary_recording
+
+import json
+import numpy as np
+from pathlib import Path
+from tqdm import tqdm
+
+from .utils.mdaio import DiskReadMda, readmda, writemda64, MdaHeader
+
+
+class MdaRecording(BaseRecording):
+    extractor_name = 'MdaRecording'
+    has_default_locations = True
+    has_unscaled = False
+    installed = True  # check at class level if installed or not
+    is_writable = True
+    mode = 'folder'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, folder_path, raw_fname='raw.mda', params_fname='params.json', geom_fname='geom.csv'):
+        folder_path = Path(folder_path)
+        self._folder_path = folder_path
+        self._dataset_params = read_dataset_params(self._folder_path, params_fname)
+        self._timeseries_path = self._folder_path / raw_fname
+        geom = np.loadtxt(self._folder_path / geom_fname, delimiter=',', ndmin=2)
+        self._diskreadmda = DiskReadMda(str(self._timeseries_path))
+        dtype = self._diskreadmda.dt()
+        num_channels = self._diskreadmda.N1()
+        assert geom.shape[0] == self._diskreadmda.N1(), f'Incompatible dimensions between geom.csv and timeseries ' \
+                                                        f'file: {geom.shape[0]} <> {self._diskreadmda.N1()}'
+        BaseRecording.__init__(self, sampling_frequency=self._dataset_params['samplerate'] * 1.0,
+                               channel_ids=np.arange(num_channels), dtype=dtype)
+        rec_segment = MdaRecordingSegment(self._diskreadmda)
+        self.add_recording_segment(rec_segment)
+        self.set_channel_locations(geom)
```

Comment:
We should create a probe object here no ?

Reply:
possibly, but we don't have default electrode info. We should have a function that makes a probe out of locations. I think it should be the default behavior in set_locations